src/commands/functions/create.js

```diff
@@ -196,30 +195,78 @@ const pickTemplate = async function () {
 
 const DEFAULT_PRIORITY = 999
 
-/* get functions dir (and make it if necessary) */
-const ensureFunctionDirExists = function (context, flags, config) {
-  const functionsDir = config.build && config.build.functions
-  if (!functionsDir) {
-    context.log(`${NETLIFYDEVLOG} No functions folder specified in netlify.toml`)
-    process.exit(1)
+/**
+ * Get functions directory (and make it if necessary)
+ * @param {FunctionsCreateCommand} context
+ * @returns {string | never} - functions directory or throws an error
+ */
+const ensureFunctionDirExists = async function (context) {
+  const { api, config, site } = context.netlify
+  const siteId = site.id
+  const { log } = context
+  let functionsDirHolder = config.build && config.build.functions
+
+  if (!functionsDirHolder) {
+    log(`${NETLIFYDEVLOG} functions directory not specified in netlify.toml or UI settings`)
+
+    if (!siteId) {
+      context.log(
+        `${NETLIFYDEVERR} Unable to find a site id. Please make sure you are in a directory that is linked to a site with Netlify.`,
```

Comment:
@klavavej This is new copy, please let me know what you think!

Reply:
Thanks for the ping @tiffanosaurus !

[pebble] We already have a few variants of this error message in the CLI, I'm reluctant to introduce a new variant so I suggest using one of the existing ones. 

Here's a suggestion to change this to [the most common similar error message](https://github.com/netlify/cli/search?q=%22No+site+id+found%2C+please+run+inside+a+site+folder+or+%60netlify+link%60%22)

```suggestion
        `${NETLIFYDEVERR} No site id found, please run inside a site folder or `netlify link``,
```

Other similar error messages include 

> Could not find the site ID. Please run netlify link.

> Folder is not linked to a Netlify site. Run 'netlify link' to link it

But those are used far less.